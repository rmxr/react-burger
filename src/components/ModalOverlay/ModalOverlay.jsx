import React from "react";
import styles from "./ModalOverlay.module.css"

function ModalOverlay({onClose}) {
    return (
        <div onClick={onClose} className={styles.overlay}>

        </div>
    )
};

export default ModalOverlay;