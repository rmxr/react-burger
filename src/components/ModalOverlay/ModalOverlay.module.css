.overlay {
    position: fixed;
    content: "";
    width: 100%;
    height: 100%;
    top: 0;
    background-color: #000000;
    opacity: 60%;
}