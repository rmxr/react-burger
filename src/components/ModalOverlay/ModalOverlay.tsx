import React from "react";
import styles from "./ModalOverlay.module.css";

function ModalOverlay({onClose}: { onClose: () => void }) {
  return (
    <div data-testid="overlay" onClick={onClose} className={styles.overlay}>

    </div>
  )
}

export default ModalOverlay;