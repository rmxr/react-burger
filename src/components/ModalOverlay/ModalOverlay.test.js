import React from 'react';
import {render, fireEvent, screen} from '@testing-library/react';
import ModalOverlay from './ModalOverlay';
import renderer from "react-test-renderer";

test('clicking on overlay calls onClose function', () => {
  const onCloseMock = jest.fn();
  render(<ModalOverlay onClose={onCloseMock}/>);
  const overlay = screen.getByTestId('overlay');
  fireEvent.click(overlay);
  expect(onCloseMock).toHaveBeenCalled();
});

it('modalOverlay renders without mistakes', () => {
  const onCloseMock = jest.fn();
  const tree = renderer
    .create(<ModalOverlay onClose={onCloseMock}/>)
    .toJSON();
  expect(tree).toMatchSnapshot();
})