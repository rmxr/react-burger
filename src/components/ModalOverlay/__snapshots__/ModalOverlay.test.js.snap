// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`modalOverlay renders without mistakes 1`] = `
<div
  className="overlay"
  data-testid="overlay"
  onClick={[MockFunction]}
/>
`;
